homeassistant/components/mqtt/__init__.py

```diff
@@ -181,13 +194,13 @@ def mqtt_topic_subscriber(event):
                          event.data[ATTR_PAYLOAD], event.data[ATTR_QOS],
                          priority=JobPriority.EVENT_CALLBACK)
 
-    remove = hass.bus.listen(EVENT_MQTT_MESSAGE_RECEIVED,
-                             mqtt_topic_subscriber)
+    async_remove = hass.bus.async_listen(EVENT_MQTT_MESSAGE_RECEIVED,
+                                         mqtt_topic_subscriber)
 
     # Future: track subscriber count and unsubscribe in remove
     MQTT_CLIENT.subscribe(topic, qos)
```

Comment:
Actually, it's non-blocking: https://github.com/eclipse/paho.mqtt.python/blob/master/src/paho/mqtt/client.py#L839

So its fine to call immediately.